v3/pb.go

```diff
@@ -586,3 +594,25 @@ func (s *State) IsFirst() bool {
 func (s *State) Time() time.Time {
 	return s.time
 }
+
+
+type Progressable interface {
```

Comment:
I don't understand how it may be helpful. Can you give some examples?

Reply:
So I'm using it here https://github.com/cbehopkins/medorg/blob/master/mdbackup/main.go in registerFunc.
Apologies, this code is in need of a major refactor!

Basically it means I can just point progress bar at the thing I want to have a status on its progress and it sorts itself out. I've decoupled my already overloaded main from having to have functionality in there to monitor a thing. It also means I do not directly couple pool (something very specific to that command line tool) into the library whose job it is to walk through a certain file structure. 
I was planning to use this in other places for other things that consume time, it just saves me having to keep repeating the code that queries a thing. Eventually I was again planning to refactor this type to be a variation on progressBar, so that instead of a progressable querying a thing and updating an underlying progress bar, that the progressBar's worker then queried, instead to have the variant of the prgressBar's worker query the thing directly.
But again I wondered if you thought this was sensible - no point me going too far down a rabbit hole.

Reply:
I like your example it's original and awesome to use pb in this way :) That something about I think when made v3.
But for me, progressWorker not looks like a common and useful feature and I think your repo will be the best place for it. We need to stay as just a functional progress bar.